x/evm/types/ethtx/utils.go

```diff
@@ -0,0 +1,34 @@
+package ethtx
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+// Gas price is the smaller of base fee + tip limit vs total fee limit
+func EffectiveGasPrice(baseFee, feeCap, tipCap *big.Int) *big.Int {
+	return math.BigMin(new(big.Int).Add(tipCap, baseFee), feeCap)
+}
+
+func MustSetConvertIfPresent[U comparable, V any](orig U, converter func(U) (V, error), setter func(V)) {
```

Comment:
Can we add a comment for what this is doing?

Reply:
removed it altogether